thunder/dynamo/benchmark_utils.py

```diff
@@ -375,8 +385,28 @@ def check_timing(
     and generate a benchmark script if the difference exceeds the threshold.
     """
     graph_name = report.graph_name
-    measure1 = report.run_benchmark(compile_fn1, timer_fn)
-    measure2 = report.run_benchmark(compile_fn2, timer_fn)
+    if not is_report_using_cuda(report):
+        print(f"{graph_name} doesn't use CUDA, skip benchmark {timer_name}")
+        return
+    filename1 = f"{graph_name}_{compile_fn1.name}_{timer_name}.py"
+    filename2 = f"{graph_name}_{compile_fn2.name}_{timer_name}.py"
+
+    def try_and_log_benchmark(compile_fn, filename):
+        try:
+            return report.run_benchmark(compile_fn, timer_fn)
+        except Exception as e:
+            msg = f"Benchmark {timer_name} on {graph_name} using {compile_fn.name} failed with exception {e}, benchmark script failed_{filename} is saved"
+            print(msg)
+            report.write_benchmark(
+                folder_path, compile_fn, timer_fn, file_name=f"failed_{filename1}", extra_comment_str=msg
```

Comment:
Nice filename that indicates what happened

In the future we may want to think about creating a special folder for failures